scylla-gdb.py

```diff
@@ -838,18 +838,29 @@ def _lookup_type(type_name):
         types = [_lookup_type('sstables::sstable_mutation_reader<sstables::data_consume_rows_context_m, sstables::mp_row_consumer_m>'),
                  _lookup_type('sstables::sstable_mutation_reader<sstables::data_consume_rows_context, sstables::mp_row_consumer_k_l>')]
 
-    def _lookup_obj(obj_addr, vtable_addr):
+    def _lookup_obj(obj_addr, vtable_addr, vtable_addr_to_ptr_type):
+        try:
+            # lookup cache
+            return vtable_addr_to_ptr_type[vtable_addr]
+        except KeyError:
+            pass
+        # insert negative entry in cache in case
+        # vtable_addr is not found or not matching any reader type
+        vtable_addr_to_ptr_type[vtable_addr] = None
```

Comment:
I think this can make `vtable_addr_to_ptr_type` grow to O(core dump size), which may overflow host's memory.

Rather than optimizing this, we should do a code change similar to 33bef82f6bb30e6623b84a18a2ac245f3292042a

Reply:
I suggested this optimization because it greatly reduced the runtime of another (similar) command that I wrote. However I just found out today, that if there are millions of virtual objects in memory (happens often when scylla dumps core), the command still takes hours to run becoming effectively useless :(.
So I agree with Tomek, commands whose runtime scales with the number of virtual objects will be of limited usefullness at best.

I think we can go one step further and link all readers into an intrusive list. Filtering through that list will be much faster than filtering all virtual objects.

Reply:
I was under the (false) assumption that vtable_addr is O(#types) rather than O(#objects)...

I'll resend just the first, unoptimized, patch then and later we can better track the readers in scylla as Botond suggested.

I see that we have the same scalability issue in `resolve` with addr -> name mapping and it has a `cache` param to enable the cache. We can add a flag to `scylla active sstables` or even a global flag (e.g. `scylla set caching [on|off]` command) that will control default metadata caching in scylla-gdb so we could turn it off if we run out of memory due to having too many objects.

Reply:
`vtable_addr` *is* `O(#types)`. It is the scan of the entire memory that is `O(#objects)`. Caching speeds that up to some extent but if there are millions of objects the scan still takes a lot of time (hours even).

Reply:
Yes. this pr is not attempting to fix that.
I suggest we merge this one since it does improve functionality until we do the intrusive list of sstable readers, and do the latter as a follow up.

Reply:
Agree.